recipes/newrelic/infrastructure/ohi/nginx/debian.yml

```diff
@@ -40,49 +40,95 @@ validationNrql: "SELECT count(*) from NginxSample where hostname like '{{.HOSTNA
 successLinkConfig:
   type: EXPLORER
 
-inputVars:
-  - name: "NR_CLI_STUB_STATUS_URL"
-    prompt: "NGINX status URL (default: http://127.0.0.1/status)"
-    default: "http://127.0.0.1/status"
-
 preInstall:
   info: |2
       To capture data from the NGINX integration, you'll first need to enable
       and configure the applicable extension module:
       - For NGINX Open Source: HTTP stub status module 
       - For NGINX Plus: HTTP status module and HTTP API module
+      - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/nginx-monitoring-integration#config for more info.
 
 install:
   version: "3"
   silent: true
-
   tasks:
     default:
       cmds:
+        - task: input
+          vars:
+            MAX_RETRIES: 3
         - task: assert_pre_req
+          vars:
+            NR_CLI_STUB_STATUS_URL:
+              sh: sudo cat /tmp/nginx-ohai-input
         - task: setup
+          vars:
+            NR_CLI_STUB_STATUS_URL:
+              sh: sudo cat /tmp/nginx-ohai-input
+            NR_CLI_STATUS_MODULE:
+              sh: sudo cat /tmp/nginx-ohai-status-module
         - task: restart
 
+    input:
+      cmds:
+        - |
+          tries=0
+          while [ $tries -lt {{.MAX_RETRIES}} ]; do
+            printf "\nNGINX status URL (default: http://127.0.0.1/status): "
+            read -r input
+            input=${input:-http://127.0.0.1/status}
+            code=$(curl -I $input -w %{response_code} -so '/dev/null')
+            if [ $code == "200" ]; then
+              echo $input > /tmp/nginx-ohai-input
+              chmod 600 /tmp/nginx-ohai-input && sudo chown root:root /tmp/nginx-ohai-input
```

Comment:
This workaround saving the input data into a temporary file has been the only way we've found with go-task to pass dynamic data between tasks, because, so far, they don't have another way as stated in :
- https://github.com/go-task/task/issues/92
- https://github.com/go-task/task/pull/236

Reply:
The file is set with only root reading permissions, it's removed after use, and, if, it would be a sensitive url with user and password, at the end that information is gonna be in the config.yml file anyway.